core/components/commerce_itemoptions/src/Modules/ItemOptions.php

```diff
@@ -94,7 +97,7 @@ public function onAddToCart(Item $event): void
             ]);
             $item->addPriceAdjustment($adjustment);
         }
-
+        $item->setProperty('item_childsku', join("-", $skuList));
```

Comment:
This may be problematic because a product SKU can contain a hyphen.

Reply:
So what do you suggest as separator?

Reply:
Or we can exclude adding the same in item properties.

Reply:
Yeah, I'd prefer this not added to the `comOrderItem` properties. Perhaps I'm missing something; is there a particular reason you've added it? 

Referencing your first commit, you'd still be able to access the item option SKUs in the twig templates via the adjustments field. e.g.
```
{% for item in items %}
    {% for adjustment in item.adjustments %}
        {{ adjustment.properties.itemopt_product.sku }}
    {% endfor %}
{% endfor %}
```

BTW thanks for contributing! 🙂

Reply:
The reason: In e-commerce it's standard practice to generate parent product SKU using it's options SKU by gluing using hyphen, so i did. I also do Magento 2 so very well aware about it's practices w.r.t. SKUs.

In my current usecase it's not useful but i thought it may be helpful to community, also if it's already there then we won't have to run 2 loops to print such SKU.

So keeping it will also not harm anything i think.

Still if you want i can remove it right away.

Reply:
>The reason: In e-commerce it's standard practice to generate parent product SKU using it's options SKU by gluing using >hyphen, so i did. I also do Magento 2 so very well aware about it's practices w.r.t. SKUs.

Ah right, Commerce does the same when using a Product Matrix TV, but in that case it's generating the actual product SKU rather than adding a value to the properties. My worry is that developers may use this assuming they can explode() this, which works until a user creates an incompatible SKU.

>In my current usecase it's not useful but i thought it may be helpful to community, also if it's already there then we won't >have to run 2 loops to print such SKU.

I guess it's similar to the argument concerning normalisation/denormalisation of db tables. In this case, it would typically be used inside the items loop anyway, so it's really just looping over a small adjustments array which would be negligible. 

If the second commit is reverted, then this has my vote for merging. Thanks a lot, @ankurvr!

Reply:
@muzzwood Now you can merge the PR. I just removed extra code related to parent product sku building/